### PR TITLE
Add documented option for default handshake version and update README.md

### PR DESCRIPTION
**Pull request by kjellwinblad**
* Add documented option for setting default handshake version
* Improve description in the README.md file and add description for how to use and test brod_gssapi

This PR attempts to solve:

* https://github.com/kafka4beam/brod_gssapi/issues/10 and
* https://github.com/kafka4beam/brod_gssapi/issues/11.

### CODE REVIEW

File: src/brod_gssapi.erl [hunk 1]
```diff
@@ -17,7 +17,7 @@
     mechanism := binary(),
     sasl_context := binary(),
     sasl_conn := sasl_auth:state() | undefined,
-    handshake_vsn := non_neg_integer() | undefined
+    handshake_vsn := non_neg_integer() | legacy
```
Comment (starbelly) [inline src/brod_gssapi.erl:20]: Much better :) 💯

**Review by starbelly — APPROVED** [review]
Beautiful 👍

File: README.md [hunk 2]
```diff
@@ -1,11 +1,66 @@
...
+handshake versions that are supported. `brod_gssapi` uses
+[sasl_auth](https://github.com/ElMaxo/sasl_auth), which is an Erlang wrapper
+for a SASL/GSSAPI C library. 
 
-Build
+Usage
+-----
+
+1. Install the dependencies for `sasl_auth`. More information about
+   `sasl_auth`'s dependencies can be found in [`sasl_auth`'s README.md file](https://github.com/ElMaxo/sasl_auth).
```
Comment (zmstone) [inline README.md:17]: Sorry for the late review.
Here should update to new url.

File: README.md [hunk 3]
```diff
@@ -1,11 +1,66 @@
...
+and a Brod client. The `example/README.md` file describes how to run this
+example. The code in `example/brod_client/src/example.erl` sets up a Brod
+client with SASL/GSSAPI (Kerberos) authentication and sends and receives
+messages. 
+
+
+Dependencies
+------------
+
+Please see [`sasl_auth`'s README.md file](https://github.com/ElMaxo/sasl_auth)
```
Comment (zmstone) [inline README.md:34]: New url

Comment (kjellwinblad) [reply README.md:34]: Thanks @zmstone, I will fix these two URLs.

Comment (kjellwinblad) [reply README.md:34]: Fixed here aaa3da5f8132408d249d8223688d00d8627d4d55

**Pull request merged**